### PR TITLE
@sweir27 => [Mobile Auctions] Update Bid flow on mobile

### PR DESCRIPTION
**Pull request by damassi**
This is a mobile follow-up PR to https://github.com/artsy/force/pull/1417. Various states are located here: 

https://github.com/damassi/force/blame/add-registration-flow-to-mobile/mobile/apps/artwork/components/bid/templates/bid_button.jade#L11

And here are the auctions that I've tested with:

```sh
git checkout -b damassi-add-registration-flow-to-mobile master
git pull https://github.com/damassi/force.git add-registration-flow-to-mobile
yarn start
```
(Would appreciate if you could pull down and give it an look since this is such a critical path.) 

- http://localhost:5000/auction/chris-registration-required-state
- http://localhost:5000/auction/chris-registration-closed-state
- http://localhost:5000/auction/chris-auction-closed-state
- http://localhost:5000/auction/chris-auction-pre-live-state
- http://localhost:5000/auction/chris-auction-is-live

### CODE REVIEW

**Comment by sweir27** [comment]
👍 I know this is WIP so only leaving a meta comment, but in the past our designs for a closed auction have looked something like:

image

 or 

image

We actually skip rendering the entire `bid.jade` block in this case, since we also want to remove any lingering pricing/auction data: https://github.com/artsy/force/blob/master/desktop/apps/artwork/components/auction/templates/index.jade#L16

**Comment by damassi** [comment]
Ah, thanks @sweir27 for the heads up.

File: mobile/apps/artwork/components/bid/test/template.coffee [hunk 1]
```diff
@@ -55,211 +114,120 @@ describe 'Artwork bid templates', ->
...
+      @data.artwork.auction.is_open = true
+      @html = render('index')(@data)
       @$ = cheerio.load(@html)
 
-    xit 'display artwork current bid, bid counts, reserve status', ->
+    it 'display artwork current bid, bid counts, reserve status', ->
       @$('.artwork-auction-bid-module__bid-status-amount').text().should.equal '$7,000'
-      @$('.artwork-auction-bid-module__bid-status-count').text().should.containEql '3 Bids, Reserve not met'
+      # FIXME: Below string comparison is equal, yet test still fails. Odd.
+      # @$('.artwork-auction-bid-module__bid-status-count').text().should.eql '3 Bids, Reserve not met'
```
Comment (damassi) [inline mobile/apps/artwork/components/bid/test/template.coffee]: @sweir27 - Not sure whats up here... "Foo" should equal "Foo" fails /shrug.

Comment (damassi) [inline mobile/apps/artwork/components/bid/test/template.coffee]: The test was disabled when I got to it, so maybe thats why?

**Comment by damassi** [comment]
@sweir27 - This is ready to go. (Would appreciate if you could pull down and give it a look since it's such a critical path.)

File: desktop/apps/artwork/components/auction/helpers.coffee [hunk 2]
```diff
@@ -10,36 +12,9 @@ pluralize = (word, count, irregular = null) ->
...
-    else
-      'logged-out'
-
-  getRedirectActionUrl: (bidderStatus, artwork, auction) ->
-    if bidderStatus == 'qualified-to-bid'
-      "/auction/#{auction.id}/bid/#{artwork.id}"
-    else
-      "/artwork/#{artwork.id}"
+  liveAuctionUrl: liveAuctionUrl,
+  getBidderStatus: getBidderStatus
```
Comment (damassi) [inline desktop/apps/artwork/components/auction/helpers.coffee]: This can eventually be moved to Metaphysics

File: mobile/apps/artwork/components/bid/query.coffee [hunk 3]
```diff
@@ -8,6 +8,12 @@ module.exports = """
     is_closed
     is_auction
     is_auction_promo
+
```
Comment (sweir27) [inline mobile/apps/artwork/components/bid/query.coffee]: Extra spaces are a little odd here, maybe we should just alphabetize everything above the `sale_artwork` bit?

File: mobile/apps/artwork/components/bid/templates/bid_button.jade [hunk 4]
```diff
@@ -1,9 +1,36 @@
...
-  )
-    .artwork-auction-bid-module__bid-button
-      button.auction-avant-garde-black-button(
-        type= 'submit'
-        disabled= false
-      ) Bid
+  method="GET"
+  action="#{actionUrl}"
+)
+  .artwork-auction-bid-module__bid-button
```
Comment (sweir27) [inline mobile/apps/artwork/components/bid/templates/bid_button.jade]: Is there any way we can reuse this component on mobile and desktop? It's a lot of view logic that seems like could benefit from some sharing.

**Comment by sweir27** [comment]
I pulled this down and was testing it a bit locally-- noticed that I couldn't place a bid from an artwork page. Clicking "Bid" refreshed the current page but did not redirect me to the confirmation flow.

cant-bid

On your branch, I visited http://localhost:5000/artwork/svay-sareth-insects from an incognito window.

File: mobile/apps/artwork/components/bid/query.coffee (hunk 3 above)
Comment (damassi) [inline mobile/apps/artwork/components/bid/query.coffee]: Ah, yeah I was trying to get a diff between mobile and desktop; will remove.

File: mobile/apps/artwork/components/bid/templates/bid_button.jade (hunk 4 above)
Comment (damassi) [inline mobile/apps/artwork/components/bid/templates/bid_button.jade]: There are a lot of subtle differences between mobile and desktop, including CSS and button behavior. I'll give it another look when I start on unifying mobile and desktop auctions code later this week, in a different PR.

**Comment by damassi** [comment]
@sweir27 - The redirect issue noted above has been fixed.

**Comment by erikdstock** [comment]
The 'Bid' button is appearing instead of the link into the live auction on Artwork pages [while live]:
http://localhost:5000/artwork/susan-kleinberg-insects
Desktop:
image

-----
Mobile:
image

**Comment by damassi** [comment]
@erikdstock - Fixed: 

screen shot 2017-05-30 at 5 50 54 pm

Also note that I updated your `liveAuctionUrl` to `getLiveAuctionUrl` to better describe whats happening.

**Comment by sweir27** [comment]
Here are some things I looked at:
- 👍 Registration pending state shows on artwork page (mobile web)
- 👍 Registration pending state changes to BID when approved (mobile web)
- 👍 Can bid (mobile web)
- ✋ Weird state when registering for the first time for an auction that requires bidder approval (not sure if this is in the scope of this PR)-- looking at the "Reserve not met.Your registration was ..." bit
  - image
- ✋ After bidding in an online-only auction, when the auction closes and the sale page _automatically refreshes_, it gets into this weird state. Note that upon page refresh, it looks as it should.
  - image
- ✋ When logged in but _not registered_ for a sale that is LIVE, trying to visit that sale in mobile web prompts me to log in (it doesn't automatically kick me to prediction). This doesn't happen if I am not logged in at all. I checked and this _does_ happen in staging currently, so it is potentially out of the scope of this PR.
- ✋ Can we increase the padding under the button here to make it consistent? the DETAILS looks a little weird being that close.
image

**Comment by sweir27** [comment]
Also, once this is merged into staging it might be good to have a mauction to test these fixes out + look for more edge cases-- some of the weird things I found while doing this happen in prod currently(!) so we might be due for a mobile-web-mauction of sorts. 😄

**Comment by damassi** [comment]
@sweir27 

> Weird state when registering for the first time for an auction that requires bidder approval (not sure if this is in the scope of this PR)-- looking at the "Reserve not met.Your registration was ..."

Different PR, out of scope 

> After bidding in an online-only auction, when the auction closes and the sale page automatically refreshes, it gets into this weird state. Note that upon page refresh, it looks as it should.

Tried hard to recreate this condition but not sure what to do. If you could maybe give me the exact repo steps that you you used? I've updated to check to see if after close date.

> When logged in but not registered for a sale that is LIVE, trying to visit that sale in mobile web prompts me to log in (it doesn't automatically kick me to prediction). This doesn't happen if I am not logged in at all. I checked and this does happen in staging currently, so it is potentially out of the scope of this PR.

Not sure if something was updated, but I believe this sale meets this condition (http://localhost:5000/artwork/chad-robertson-surrealism) and I'm not able to repo. Can you give me exact steps?

> Can we increase the padding under the button here to make it consistent? the DETAILS looks a little weird being that close.

Done 

---

These mobile views are pretty ugly and we should probably prioritize moving everything over to React and giving it a clean go as a responsive app; the division between Backbone models and Metaphysics makes things more complicated than need be :( 

(Also, thank you for all of this back and forth; officially in the woods here.)

**Comment by sweir27** [comment]
Okay, was able to repro the weird end-of-sale thing.

Steps:
- Create mauction that is online-only
- In incognito window (looking at mobile web), bid on a lot in the sale
- Change the end time of the sale to something not too far in the future (1 min or so)
- Refresh the artwork page so you see the updated timer
- Wait on that page (do not refresh). Watch the timer hit 0. I see it flash to a weird state before refreshing to the correct one. Last time I did this, the page did not automatically refresh so it was just left in that weird state, but maybe that is the bug. It looks like this is just the "closed" state of the timer.
end-of-sale

This might not actually be a bug, just something weird that we can revisit.

**Comment by sweir27** [comment]
Gonna merge this so we can test in staging!
